### ISSUE TITLE
User cannot access identities upon class reload

### ISSUE THREAD

**Issue by kjohnston** [newleaders/thincloud-authentication#19]
While `thincloud-authentication` provides an initializer to include this module within the application-level User model, we're having to include it again in that model to avoid a class reloading issue where `identities` will not be defined.

``` ruby
include Thincloud::Authentication::IdentifiableUser
```

@elskwid  was able to reproduce it as follows, given a `by_name` scope on the User model that joins to identities:

``` ruby
def self.by_name
  includes(:identities).order("thincloud_authentication_identities.name")
end
```

```
1.9.3p327 :002 > User.by_name
  SQL (1.9ms)  SELECT "users"."id" AS t0_r0, "users"."time_zone" AS t0_r1, "users"."created_at" AS t0_r2, "users"."updated_at" AS t0_r3, "thincloud_authentication_identities"."id" AS t1_r0, "thincloud_authentication_identities"."user_id" AS t1_r1, "thincloud_authentication_identities"."provider" AS t1_r2, "thincloud_authentication_identities"."uid" AS t1_r3, "thincloud_authentication_identities"."name" AS t1_r4, "thincloud_authentication_identities"."email" AS t1_r5, "thincloud_authentication_identities"."password_digest" AS t1_r6, "thincloud_authentication_identities"."verification_token" AS t1_r7, "thincloud_authentication_identities"."verified_at" AS t1_r8, "thincloud_authentication_identities"."created_at" AS t1_r9, "thincloud_authentication_identities"."updated_at" AS t1_r10 FROM "users" LEFT OUTER JOIN "thincloud_authentication_identities" ON "thincloud_authentication_identities"."user_id" = "users"."id" ORDER BY thincloud_authentication_identities.name
 => [#<User id: 1, time_zone: nil, created_at: "2013-03-25 01:10:14", updated_at: "2013-03-25 01:10:14">]
1.9.3p327 :003 > reload!
Reloading...
 => true
1.9.3p327 :004 > User.by_name
ActiveRecord::ConfigurationError: Association named 'identities' was not found; perhaps you misspelled it?
...
```

**Comment by elskwid** [comment]
Thanks for putting this issue in @kjohnston. I'm guessing this is complicated by using the Engine to include the `:identities` association. I'll see if @phlipper and I can take a look this while we're at MWRC.

**Comment by elskwid** [comment]
This has been fixed in #27. Thanks @phlipper!
